cdk/src/open-data-platform/data-plane/data-import/handler-factory.ts

```diff
@@ -49,23 +49,46 @@ export const geoJsonHandlerFactory =
   (s3Params: AWS.S3.GetObjectRequest, callback: ProcessCallback): ProcessHandler =>
   async (event: ProcessRequest): Promise<APIGatewayProxyResult> => {
     // Use arguments or defaults.
-    const rowOffset = event.rowOffset ?? 0;
+    let rowOffset = event.rowOffset ?? 0;
     const rowLimit = event.rowLimit ?? Infinity;
     const batchSize = event.batchSize ?? 10;
-    console.log('Starting import:', { rowLimit, rowOffset, batchSize, s3Params });
+    console.log('Starting import:', {
+      rowLimit,
+      rowOffset,
+      batchSize,
+      s3Params,
+    });
 
     const db = new AWS.RDSDataService();
     let results: ProcessResult = {
       processedBatchCount: 0,
-      sucessfulBatchCount: 0,
+      successfulBatchCount: 0,
       erroredBatchCount: 0,
     };
 
-    try {
-      results = await readGeoJsonFile(db, s3Params, callback, rowOffset, rowLimit, batchSize);
-    } catch (error) {
-      console.log(`Error after processing ${results.processedBatchCount} batches:`, error);
-      throw error;
+    let keepReading = true;
+    while (keepReading) {
```

Comment:
I added a description to the "Set up automatic DB updates" task as follows: Ideal workflow is 1 s3 trigger on all files that launch a single lambda that splits up the file into reasonably sized chunks and then writes those files to a new S3 bucket. Then triggers on those new S3 files to run individual lambda functions. 

This would mean we should not have to invoke the lambdas multiple times bc files should be small enough to be parsable. I also removed my white loop because some of the numbers I was getting from results were not making sense

Reply:
SGTM.